src/editor/components/modals/GeoModal/GeoModal.component.jsx

```diff
@@ -0,0 +1,151 @@
+import { useState, useCallback, useEffect } from 'react';
+
+import styles from './GeoModal.module.scss';
+import { Copy32Icon, Mangnifier20Icon } from '../../../icons';
+
+import { useAuthContext } from '../../../contexts/index.js';
+import Modal from '../Modal.jsx';
+import { Button, Input } from '../../components/index.js';
+import { GoogleMap, useJsApiLoader, Marker } from '@react-google-maps/api';
+import { DownloadIcon } from '../../../icons/icons.jsx';
+import GeoImg from '../../../../../ui_assets/geo.png';
+
+const GeoModal = ({ isOpen, onClose }) => {
+  const { currentUser } = useAuthContext();
+  const isProUser = currentUser && currentUser.isBeta;
+  const [markerPosition, setMarkerPosition] = useState({
+    lat: 0,
+    lng: 0,
+    elevation: 0
+  });
+
+  const roundToSix = (num) => {
+    return Math.round(num * 1e6) / 1e6;
+  };
```

Comment:
You also have that roundToSix function in another file, don't you have a utils file somewhere to put that?

Reply:
@Algorush why are we rounding down?

Reply:
If we do not round, we get very long coordinate values. And rounding to 6 digits allows to achieve an error of 0.1 meter

Reply:
without rounding: 33.95521700914383, -118.32591788277125

Reply:
Thanks for the context. The goal is 1cm precision. Let's round to 7?

Reply:
Ok!